### PR TITLE
SCRUM-1320 Switching of disease annotation enums to VocabularyTerms

### PR DESCRIPTION
**Pull request by markquintontulloch**
Need to load vocabularies and vocabulary terms before merging

### CODE REVIEW

File: src/main/java/org/alliancegenome/curation_api/interfaces/crud/VocabularyCrudInterface.java [hunk 1]
```diff
@@ -28,4 +28,9 @@
     @JsonView(View.VocabularyTermView.class)
     ObjectListResponse<VocabularyTerm> getTerm(@PathParam("id") Long id);
     
+    @GET
+    @Path("/findBy/{name}")
+    @JsonView(View.FieldsAndLists.class)
+    public ObjectResponse<Vocabulary> get(@PathParam("name") String name);
```
Comment (oblodgett) [inline src/main/java/org/alliancegenome/curation_api/interfaces/crud/VocabularyCrudInterface.java]: I think is would be good to change the method name from `get` to `findByName` it just makes it more clear.

File: src/main/java/org/alliancegenome/curation_api/controllers/crud/VocabularyCrudController.java [hunk 2]
```diff
@@ -26,6 +26,16 @@ protected void init() {
...
+        SearchResponse<Vocabulary> ret = findByField("name", name);
+        if(ret != null && ret.getTotalResults() == 1) {
+            return new ObjectResponse<>(ret.getResults().get(0));
+        } else {
+            return new ObjectResponse<>();
+        }
+    }
 
     @Override
     public ObjectListResponse<VocabularyTerm> getTerm(Long id) {
```
Comment (oblodgett) [inline src/main/java/org/alliancegenome/curation_api/controllers/crud/VocabularyCrudController.java]: This method `getTerm` should be renamed to `getTerms` I know thats not part of this PR but anyway.

File: src/main/java/org/alliancegenome/curation_api/services/DiseaseAnnotationFmsService.java [hunk 3]
```diff
@@ -43,6 +42,12 @@
     @Inject ExperimentalConditionDAO experimentalConditionDAO;
     @Inject BiologicalEntityDAO biologicalEntityDAO;
     @Inject GeneDAO geneDAO;
+    @Inject VocabularyTermDAO vocabularyTermDAO;
+    
+    private String CONDITION_RELATION_TYPE_VOCABULARY = "Condition relation types";
+    private String GENE_DISEASE_RELATION_VOCABULARY = "Gene disease relations";
+    private String ALLELE_DISEASE_RELATION_VOCABULARY = "Allele disease relations";
+    private String AGM_DISEASE_RELATION_VOCABULARY = "AGM disease relations";
```
Comment (oblodgett) [inline src/main/java/org/alliancegenome/curation_api/services/DiseaseAnnotationFmsService.java]: These would probably be better pulled out into a `Constants` class. And typically they are defined as:

`private static final String CONSTANT = "CONSTANT";`

File: src/main/java/org/alliancegenome/curation_api/services/DiseaseAnnotationService.java [hunk 4]
```diff
@@ -62,6 +58,14 @@
...
     @Inject
     BiologicalEntityDAO biologicalEntityDAO;
+    @Inject
+    VocabularyTermDAO vocabularyTermDAO;
+    
+    private String ANNOTATION_TYPE_VOCABULARY = "Annotation types";
+    private String DISEASE_QUALIFIER_VOCABULARY = "Disease qualifiers";
+    private String DISEASE_GENETIC_MODIFIER_RELATION_VOCABULARY = "Disease genetic modifier relations";
+    private String GENETIC_SEX_VOCABULARY = "Genetic sexes";
+    private String CONDITION_RELATION_TYPE_VOCABULARY = "Condition relation types";
```
Comment (oblodgett) [inline src/main/java/org/alliancegenome/curation_api/services/DiseaseAnnotationService.java]: More constants.

File: src/main/java/org/alliancegenome/curation_api/services/helpers/validators/GeneDiseaseAnnotationValidator.java [hunk 5]
```diff
@@ -21,6 +20,10 @@
     AffectedGenomicModelDAO agmDAO;
     @Inject
     GeneDiseaseAnnotationDAO geneDiseaseAnnotationDAO;
+    @Inject
+    VocabularyTermDAO vocabularyTermDAO;
+    
+    private String GENE_DISEASE_RELATION_VOCABULARY = "Gene disease relations";
```
Comment (oblodgett) [inline src/main/java/org/alliancegenome/curation_api/services/helpers/validators/GeneDiseaseAnnotationValidator.java:26]: Now I am seeing that they are defined in multiple classes. They definitely need to get pulled out into a class of their own.

File: src/test/java/org/alliancegenome/curation_api/bulkupload/DiseaseAnnotationBulkUploadITCase.java [hunk 6]
```diff
@@ -47,9 +48,15 @@
...
-    private ArrayList<String> requiredGenes = new ArrayList<String>(Arrays.asList( "DATEST:Gene0001", "DATEST:Gene0002", "HGNC:0001"));
+    private String requiredGeneDiseaseRelation = "is_implicated_in";
+    private String requiredAlleleDiseaseRelation = "is_implicated_in";
+    private String requiredAgmDiseaseRelation = "is_model_of";
+    private String requiredGeneticSex = "male";
+    private String requiredDiseaseGeneticModifierRelation = "ameliorated_by";
+    private String requiredDiseaseQualifier = "susceptibility";
+    private String requiredAnnotationType = "manually_curated";
+    private String requiredNoteType = "disease_summary";
+    private String requiredConditionRelationType = "exacerbated_by";
```
Comment (oblodgett) [inline src/test/java/org/alliancegenome/curation_api/bulkupload/DiseaseAnnotationBulkUploadITCase.java:59]: These would probably do well in a Constants class also.

File: src/main/java/org/alliancegenome/curation_api/services/helpers/validators/AGMDiseaseAnnotationValidator.java [hunk 7]
```diff
@@ -38,7 +42,7 @@ public AGMDiseaseAnnotation validateAnnotation(AGMDiseaseAnnotation uiEntity) {
         AffectedGenomicModel subject = validateSubject(uiEntity, dbEntity);
         if(subject != null) dbEntity.setSubject(subject);
 
-        DiseaseRelation relation = validateDiseaseRelation(uiEntity, dbEntity);
```
Comment (oblodgett) [inline src/main/java/org/alliancegenome/curation_api/services/helpers/validators/AGMDiseaseAnnotationValidator.java:41]: Did the DiseaseRelation enum get cleaned up? Didn't see the delete of that code?

**Review by oblodgett — CHANGES_REQUESTED** [review]

File: src/main/java/org/alliancegenome/curation_api/interfaces/crud/VocabularyCrudInterface.java (hunk 1 above)
Comment (markquintontulloch) [reply src/main/java/org/alliancegenome/curation_api/interfaces/crud/VocabularyCrudInterface.java]: Done

File: src/main/java/org/alliancegenome/curation_api/controllers/crud/VocabularyCrudController.java (hunk 2 above)
Comment (markquintontulloch) [reply src/main/java/org/alliancegenome/curation_api/controllers/crud/VocabularyCrudController.java]: Done

File: src/main/java/org/alliancegenome/curation_api/services/DiseaseAnnotationFmsService.java (hunk 3 above)
Comment (markquintontulloch) [reply src/main/java/org/alliancegenome/curation_api/services/DiseaseAnnotationFmsService.java]: Moved to VocabularyConstants class

File: src/main/java/org/alliancegenome/curation_api/services/DiseaseAnnotationService.java (hunk 4 above)
Comment (markquintontulloch) [reply src/main/java/org/alliancegenome/curation_api/services/DiseaseAnnotationService.java]: Also moved to VocabularyConstants class

File: src/main/java/org/alliancegenome/curation_api/services/helpers/validators/GeneDiseaseAnnotationValidator.java (hunk 5 above)
Comment (markquintontulloch) [reply src/main/java/org/alliancegenome/curation_api/services/helpers/validators/GeneDiseaseAnnotationValidator.java:26]: Yep, done

File: src/test/java/org/alliancegenome/curation_api/bulkupload/DiseaseAnnotationBulkUploadITCase.java (hunk 6 above)
Comment (markquintontulloch) [reply src/test/java/org/alliancegenome/curation_api/bulkupload/DiseaseAnnotationBulkUploadITCase.java:59]: @oblodgett - I'm less inclined to move these.  They are specific to the test files used for the tests in that class and I think they are easier to maintain if kept in the test class.

**Review by oblodgett — APPROVED** [review]
